pybossa/sched.py

```diff
@@ -267,16 +270,32 @@ def template_get_locked_task(project_id, user_id=None, user_ip=None,
         sql = query_factory(project_id, user_id=user_id, limit=limit,
                             rand_within_priority=rand_within_priority,
                             task_type=task_type)
+        db_max_records = 100000
```

Comment:
Looks like there is likelihood of fetching huge amount of data to process from database when there are many tasks available and code [here](https://github.com/bloomberg/pybossa/pull/595/files#diff-f7eb1704644fb813492849abfa5e3f07f4c3cd76c4f60088485e5ffe204b5024R295) could run 100k times that was earlier could run max 5 times? Is this expected behavior that server could hang depending on available task data a project has?

Reply:
Yes, the code could run 100k times and it's expected behavior. The server might be slowed down a bit but 100k is a constant value so I don't think it will be hang up. 
Do you have any suggestions on speed up the process?

Reply:
This code also means that server could potentially return 100k rows instead of 5 rows that seems very resource(cpu, memory) intensive both on database and pybossa server side. My suggestion would be going for a smaller set similar to what it was earlier ~5.

Reply:
While I cannot import 100k tasks, I have the for loop run multiple times so that the code is executed >100k times. 
```responding time of requesting new task is 0:00:00.293665, task scaned: 100763``` Is this time acceptable?

Reply:
As discussed, this can be made configurable with less records(~10k instead of 100k), monitor the resource usage on dev/prod and continue investigation under future PR to improvise the behavior under which task ranking can be done offline so that ranking data would be readily available when presenting task, thereby avoid loading records around same as before(5 records). Thanks!